Sources/SwiftProcessing/Core/SketchSettings.swift

```diff
@@ -1,42 +1,188 @@
-    var strokeWeight: Double = 1
-
-    func restore(sketch: Sketch) {
-        sketch.textAlign(textAlignment)
-        sketch.textSize(textSize)
-        sketch.textFont(textFont)
-        sketch.ellipseMode(ellipseMode)
-        sketch.rectMode(rectMode)
-        sketch.imageMode(imageMode)
-        sketch.fill(fill)
-        sketch.stroke(stroke)
-        sketch.strokeWeight(strokeWeight)
-    }
-    
-    func debugPrintSettings() {
-        print("""
-            textSize = \(textSize)
-            textFont = \(textFont)
-            textAlignment = \(textAlignment)
-            ellipseMode = \(ellipseMode)
+public extension Sketch{
+
+    struct SketchSettings {
+        // Source: https://processing.org/reference/push_.html
+        // push() stores information related to the current transformation state and style settings controlled by the following functions: rotate(), translate(), scale(), fill(), stroke(), tint(), strokeWeight(), strokeCap(), strokeJoin(), imageMode(), rectMode(), ellipseMode(), colorMode(), textAlign(), textFont(), textMode(), textSize(), textLeading().
+        // The source code has a more comprehensive list of states.
+        
+        var colorMode = Default.colorMode
+        var fill = Default.fill
+        var stroke = Default.stroke
+        var tint = Default.tint
+        var strokeWeight = Default.strokeWeight
+        var strokeJoin = Default.strokeJoin
+        var strokeCap = Default.strokeCap
+        var rectMode = Default.rectMode
+        var ellipseMode = Default.ellipseMode
+        var imageMode = Default.imageMode
+        var textFont = Default.textFont
+        var textSize = Default.textSize
+        var textLeading = Default.textLeading
+        var textAlign = Default.textAlign
+        var textAlignY = Default.textAlignY
+        var blendMode = Default.blendMode
+        
+        // Re: textMode – A bitmap mode could be done using CTLineDraw from Core Text.
+        // Source 1: https://developer.apple.com/documentation/coretext/1511145-ctlinedraw
+        // Source 2: https://blog.krzyzanowskim.com/2020/07/13/coretext-swift-academy-part-3/
+        
+        static func defaultSettings(_ sketch: Sketch) {
+            sketch.settings.reapplySettings(sketch)
+        }
+        
+        public func reapplySettings(_ sketch: Sketch) {
+            sketch.colorMode(colorMode)
+            sketch.fill(fill)
+            sketch.stroke(stroke)
+            sketch.tint(tint)
+            sketch.strokeWeight(strokeWeight)
+            sketch.strokeJoin(strokeJoin)
+            sketch.strokeCap(strokeCap)
+            sketch.rectMode(rectMode)
+            sketch.ellipseMode(ellipseMode)
+            sketch.imageMode(imageMode)
+            sketch.textFont(textFont)
+            sketch.textSize(textSize)
+            sketch.textLeading(textLeading)
+            sketch.textAlign(textAlign)
+        }
+        
+        public func debugSettings() {
+            print("""
```

Comment:
smart idea! No need to change, but for future reference https://developer.apple.com/documentation/swift/customstringconvertible